src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/unsubscribe_button_request.cc

```diff
@@ -0,0 +1,157 @@
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    rpc_service::RPCService& rpc_service,
+    HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handle)
+    : RequestToHMI(message,
+                   application_manager,
+                   rpc_service,
+                   hmi_capabilities,
+                   policy_handle)
+    , EventObserver(application_manager.event_dispatcher()) {
+  hmi_apis::Common_ButtonName::eType button_name =
+      static_cast<hmi_apis::Common_ButtonName::eType>(
+          (*message_)[app_mngr::strings::msg_params]
+                     [app_mngr::strings::button_name]
+                         .asInt());
+
+  button_name_ = button_name;
+}
+
+UnsubscribeButtonRequest::~UnsubscribeButtonRequest() {}
+
+void UnsubscribeButtonRequest::Run() {
+  SDL_LOG_AUTO_TRACE();
+
+  subscribe_on_event(hmi_apis::FunctionID::Buttons_UnsubscribeButton,
+                     correlation_id());
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+
+  if (!app) {
+    SDL_LOG_ERROR("Application for connection key: " << application_id()
+                                                     << " was not found");
+    return;
+  }
+
+  app->AddPendingButtonUnsubscription(correlation_id(), button_name_);
+
+  SendRequest();
+}
+
+void UnsubscribeButtonRequest::onTimeOut() {
+  SDL_LOG_AUTO_TRACE();
+  application_manager_.updateRequestTimeout(
+      connection_key(), correlation_id(), 0);
+  auto& resume_ctrl = application_manager_.resume_controller();
+
+  resume_ctrl.HandleOnTimeOut(
+      correlation_id(),
+      static_cast<hmi_apis::FunctionID::eType>(function_id()));
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+
+  if (!app) {
+    SDL_LOG_ERROR("Application for connection key: " << application_id()
+                                                     << " was not found");
+    return;
+  }
+
+  app->RemovePendingButtonUnsubscription(correlation_id());
+}
+
+void UnsubscribeButtonRequest::on_event(const event_engine::Event& event) {
+  SDL_LOG_AUTO_TRACE();
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  if (!app) {
+    SDL_LOG_ERROR("NULL pointer.");
+    return;
+  }
+
+  const auto pending_button_unsubscriptions =
+      app->PendingButtonUnsubscriptions();
+
+  const auto it =
+      pending_button_unsubscriptions.find(event.smart_object_correlation_id());
+
+  const bool is_pending = it != pending_button_unsubscriptions.end();
+
+  hmi_apis::Common_Result::eType hmi_result =
+      static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+
+  if (hmi_apis::Common_Result::SUCCESS == hmi_result && is_pending) {
```

Comment:
Please double check do we need another success result codes here

Reply:
fixed in 2529ff7558e0a8cc1429308d019fd9e9fd7b57ca